### PR TITLE
Drop 0X and 11 support, require c++14

### PR DESCRIPTION
**Pull request by agaida**
While we are on it - just bump the requirement for cmake to 3.6.2
Bumped minor version to 6
fixes lxqt/lxqt/issues/1568

### CODE REVIEW

File: cmake/modules/ECMFindModuleHelpers.cmake [hunk 1]
```diff
@@ -126,11 +126,11 @@
 include(CMakeParseArguments)
 
 macro(ecm_find_package_version_check module_name)
-    if(CMAKE_VERSION VERSION_LESS 2.8.12)
-        message(FATAL_ERROR "CMake 2.8.12 is required by Find${module_name}.cmake")
+    if(CMAKE_VERSION VERSION_LESS 3.6.2)
+        message(FATAL_ERROR "CMake 3.6.2 is required by Find${module_name}.cmake")
```
Comment (agaida) [inline cmake/modules/ECMFindModuleHelpers.cmake]: Ok - i guess we should improve the message

File: cmake/modules/ECMFindModuleHelpers.cmake [hunk 2]
```diff
@@ -126,11 +126,11 @@
...
     endif()
-    if(CMAKE_MINIMUM_REQUIRED_VERSION VERSION_LESS 2.8.12)
-        message(AUTHOR_WARNING "Your project should require at least CMake 2.8.12 to use Find${module_name}.cmake")
+    if(CMAKE_MINIMUM_REQUIRED_VERSION VERSION_LESS 3.6.2)
+        message(AUTHOR_WARNING "Your project should require at least CMake 3.6.2 to use Find${module_name}.cmake")
```
Comment (agaida) [inline cmake/modules/ECMFindModuleHelpers.cmake]: dito

File: cmake/modules/LXQtTranslateTs.cmake [hunk 3]
```diff
@@ -63,7 +63,8 @@
 #
 
 # CMake v2.8.3 needed to use the CMakeParseArguments module
-cmake_minimum_required(VERSION 2.8.3 FATAL_ERROR)
+# We check against 3.6.2 before so this check could go away imho
+cmake_minimum_required(VERSION 3.6.2 FATAL_ERROR)
```
Comment (agaida) [inline cmake/modules/LXQtTranslateTs.cmake]: The check is not needed here imho - or is there a special reason to check again?

File: cmake/modules/LXQtCompilerSettings.cmake [hunk 4]
```diff
@@ -179,16 +179,13 @@ endif()
...
 #-----------------------------------------------------------------------------
-# CXX11 and CXX0X requirements
-#-----------------------------------------------------------------------------
-CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
-CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
-if(COMPILER_SUPPORTS_CXX11)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-elseif(COMPILER_SUPPORTS_CXX0X)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+# CXX14 requirements
```
Comment (palinek) [inline cmake/modules/LXQtCompilerSettings.cmake]: We should mess with the compiler flags by our self. Up from cmake 3.1 there is a `CXX_STANDARD` target property and the default `CMAKE_CXX_STANDARD`. We should use it together with `CXX_STANDARD_REQUIRED` (and the default `CMAKE_CXX_STANDARD_REQUIRED`).

ref. https://cmake.org/cmake/help/v3.1/prop_tgt/CXX_STANDARD.html#prop_tgt:CXX_STANDARD

Comment (agaida) [reply cmake/modules/LXQtCompilerSettings.cmake]: Agreed.

File: cmake/modules/LXQtTranslateTs.cmake (hunk 3 above)
Comment (agaida) [reply cmake/modules/LXQtTranslateTs.cmake]: erm - it's a test/set and it is needed, i also put it into the compiler settings

**Comment by agaida** [comment]
@palinek - are the changes ok?

File: cmake/modules/LXQtCompilerSettings.cmake [hunk 5]
```diff
@@ -179,16 +179,14 @@ endif()
...
+#-----------------------------------------------------------------------------
+CHECK_CXX_COMPILER_FLAG("-std=c++14" COMPILER_SUPPORTS_CXX14)
```
Comment (palinek) [inline cmake/modules/LXQtCompilerSettings.cmake]: I would just force setting the CMAKE_CXX_STANDARD(_REQUIRED) ... and that's all. No checking of compiler flags, no error message on our own. Just let everything to cmake.

File: cmake/modules/LXQtCompilerSettings.cmake [hunk 6]
```diff
@@ -179,16 +179,14 @@ endif()
...
+if(COMPILER_SUPPORTS_CXX14)
+    set(CMAKE_CXX_STANDARD_REQUIRED True)
+    set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ ISO Standard")
 else()
```
Comment (agaida) [inline cmake/modules/LXQtCompilerSettings.cmake]: @palinek - i thought i change it as you suggested?

**Comment by agaida** [comment]
Thanks - i have a problem right now - i would like to merge lxqt-transupdate and write a wrapper script around in lxqt-release-foo - the removal of UPDATE_TRANSLATIONS can wait. I will test the shiny new thing hard, not that we introduce something fishy. Tests will be in Arch and if debian is usable again in debian. What do you think?

**Comment by agaida** [comment]
@palinek just rebuild my LXQt in Arch with the new tools and Qt 5.2 beta - works without flaws

**Pull request merged**